rundeck_action/action_types/lib/slack.ts

```diff
@@ -0,0 +1,65 @@
+import { IncomingWebhook, IncomingWebhookResult } from '@slack/webhook';
+import { Logger } from '../../logger'
+
+export interface SlackApiOptions {
+  actionId: string;
+  slackWebhookUrl: string;
+  message: string;
+  logger: Logger;
+}
+
+export interface SlackApiResult {
+  status: 'ok' | 'error';
+  message?: string;
+  data?: any;
+  retry?: null | boolean | Date;
+}
+
+export async function callSlackApi(options: SlackApiOptions): Promise<SlackApiResult> {
+  const { actionId, slackWebhookUrl, message, logger } = options;
+
+  let result: IncomingWebhookResult;
+
+  try {
+
+    const webhook = new IncomingWebhook(slackWebhookUrl);
+    result = await webhook.send(message);
```

Comment:
```suggestion
    return successResult(await webhook.send(message));
```

We're not gaining anything by catching this in a local var and returning at the end of the function.